example-problems/class-removed-from-google-http-client/README.md

```diff
@@ -0,0 +1,134 @@
+This project demonstrates a hidden incompatibility between
+`com.google.api-client:google-api-client:1.27.0` and
+`com.google.cloud:google-cloud-bigtable:0.81.0-alpha`. The compilation fails due to the
+incompatibility.
+
+```
+$ mvn clean compile
+ Scanning for projects...
+...
+[ERROR] COMPILATION ERROR : 
+ -------------------------------------------------------------
+[ERROR] /usr/local/google/home/suztomo/cloud-opensource-java/example-problems/class-removed-from-google-http-client/src/main/java/com/google/cloud/tools/examples/HelloTransport.java:[31,50] cannot access com.google.api.client.http.apache.ApacheHttpTransport
+  class file for com.google.api.client.http.apache.ApacheHttpTransport not found
+ 1 error
+ -------------------------------------------------------------
+ ------------------------------------------------------------------------
+ BUILD FAILURE
+```
+
+# Diagnosis
+
+Maven artifact `com.google.api-client:google-api-client:1.27.0` contains `GoogleApacheHttpTransport`
+class.
+In this project, `HelloTransport` class tries to call
+`GoogleApacheHttpTransport.newTrustedTransport()`. This method returns `ApacheHttpTransport` class.
+This class is available in `com.google.http-client:google-http-client:1.27.0`, which is one of the
+transitive dependencies of the google-__api__-client artifact. The google-__api__-client version
+1.27.0 and the google-__http__-client version 1.27.0 work fine without any problems.
+
+However, the google-__api__-client version 1.__27__.0 and the google-__http__-client version
+1.__28__.0 has a linkage error.
+When the `pom.xml` uses `com.google.api-client:google-api-client:1.27.0` and
+`com.google.cloud:google-cloud-bigtable:0.81.0-alpha` together in this order, Maven picks up 
+`com.google.http-client:google-http-client:1.28.0` in the transitive dependencies.
+This version of the google-__http__-client artifact does not have `ApacheHttpTransport` any more
+([PR #543: Split http apache artifact](
```

Comment:
Was this a mistake? Should they not have done this? Is there a JLBP that addresses this?

Reply:
As discussed, this should have done with major version upgrade or google-http-client declaring google-http-client-apache dependency